src/toil/wdl/wdltoil.py

```diff
@@ -1578,6 +1587,46 @@ def map_over_typed_files_in_value(value: WDL.Value.Base, transform: Callable[[WD
         # All other kinds of value can be passed through unmodified.
         return value
 
+def map_over_files_in_value_check_null_type(value: WDL.Value.Base, original_value: WDL.Value.Base, expected_type: WDL.Type.Base) -> None:
+    """
+    Run through all nested values embedded in the given value and check that the null values are valid.
+
+    If a null value is found that does not have a valid corresponding expected_type, raise an error
+
+    (This is currently only used to check that coerced File types to null are valid, as in null elements have an equivalent File? type,
+    if this is to be used elsewhere, the error message should be changed to be implementation specific)
```

Comment:
```suggestion
    (This is currently only used to check that null values arising from File coercion are in locations with a nullable File? type.
    If this is to be used elsewhere, the error message should be changed to describe the appropriate types and not just talk about files.)
```

But if the function is definitely described as being about files specifically, maybe we can cut the explanation of how we might eventually generalize it to other types from the docstring?